clients/da-vinci-client/src/test/java/com/linkedin/davinci/blobtransfer/BlobSnapshotManagerTest.java

```diff
@@ -140,4 +145,42 @@ public void testSnapshotNotUpdatedWhenNotHybrid() throws RocksDBException {
     blobSnapshotManager.maybeUpdateHybridSnapshot(mockRocksDB, STORE_NAME, PARTITION_ID);
     verify(mockCheckpoint, times(0)).createCheckpoint(DB_DIR + "/.snapshot_files");
   }
+
+  @Test
+  public void testCreateSnapshotForBatch() throws RocksDBException, IOException {
+    try (MockedStatic<Checkpoint> checkpointMockedStatic = Mockito.mockStatic(Checkpoint.class)) {
+      try (MockedStatic<FileUtils> fileUtilsMockedStatic = Mockito.mockStatic(FileUtils.class)) {
+        // case 1: snapshot file not exists
+        // test prepare
+        RocksDB mockRocksDB = mock(RocksDB.class);
+        Checkpoint mockCheckpoint = mock(Checkpoint.class);
+        checkpointMockedStatic.when(() -> Checkpoint.create(mockRocksDB)).thenReturn(mockCheckpoint);
+
+        String fullSnapshotPath = DB_DIR + "/.snapshot_files";
+        BlobSnapshotManager blobSnapshotManager =
+            spy(new BlobSnapshotManager(BASE_PATH, SNAPSHOT_RETENTION_TIME, readOnlyStoreRepository));
+        File file = spy(new File(fullSnapshotPath));
+        doReturn(false).when(file).exists();
+
+        doNothing().when(mockCheckpoint).createCheckpoint(fullSnapshotPath);
+
+        // test execute
+        blobSnapshotManager.createSnapshotForBatch(mockRocksDB, fullSnapshotPath);
+        // test verify
+        verify(mockCheckpoint, times(1)).createCheckpoint(fullSnapshotPath);
```

Comment:
If the file doesn't exist, we'd like to verify the `deleteDirectory` is not called at all. Are we able to verify that?